src/_qhull_wrapper.cpp

```diff
@@ -238,103 +249,56 @@ delaunay_impl(npy_intp npoints, const double* x, const double* y,
-    const double* y;
-
-    if (!PyArg_ParseTuple(args, "O&O&",
-                          &xarray.converter_contiguous, &xarray,
-                          &yarray.converter_contiguous, &yarray)) {
-        return NULL;
-    }
+    if (x.ndim() != 1 || y.ndim() != 1)
+        throw std::invalid_argument("x and y must be 1D arrays");
 
-    npoints = xarray.dim(0);
-    if (npoints != yarray.dim(0)) {
-        PyErr_SetString(PyExc_ValueError,
-                        "x and y must be 1D arrays of the same length");
-        return NULL;
-    }
+    auto npoints = x.shape(0);
+    if (y.shape(0) != npoints)
+        throw std::invalid_argument("x and y must be 1D arrays of the same length");
 
-    if (npoints < 3) {
-        PyErr_SetString(PyExc_ValueError,
-                        "x and y arrays must have a length of at least 3");
-        return NULL;
-    }
-
-    x = xarray.data();
-    y = yarray.data();
-
-    if (!at_least_3_unique_points(npoints, x, y)) {
-        PyErr_SetString(PyExc_ValueError,
-                        "x and y arrays must consist of at least 3 unique points");
-        return NULL;
-    }
+    if (npoints < 3)
+        throw std::invalid_argument("x and y arrays must have a length of at least 3");
 
-    CALL_CPP("qhull.delaunay",
-             (ret = delaunay_impl(npoints, x, y, Py_VerboseFlag == 0)));
+    if (!at_least_3_unique_points(npoints, x.data(), y.data()))
+        throw std::invalid_argument("x and y arrays must consist of at least 3 unique points");
 
-    return ret;
+    return delaunay_impl(npoints, x.data(), y.data(), Py_VerboseFlag == 0);
 }
 
 /* Return qhull version string for assistance in debugging. */
-static PyObject*
-version(PyObject *self, PyObject *arg)
+std::string
+version()
 {
-    return PyBytes_FromString(qh_version);
+    return std::string(qh_version);
 }
 
-static PyMethodDef qhull_methods[] = {
-    {"delaunay", delaunay, METH_VARARGS,
-     "delaunay(x, y, /)\n"
-     "--\n\n"
-     "Compute a Delaunay triangulation.\n"
-     "\n"
-     "Parameters\n"
-     "----------\n"
-     "x, y : 1d arrays\n"
-     "    The coordinates of the point set, which must consist of at least\n"
-     "    three unique points.\n"
-     "\n"
-     "Returns\n"
-     "-------\n"
-     "triangles, neighbors : int arrays, shape (ntri, 3)\n"
-     "    Indices of triangle vertices and indices of triangle neighbors.\n"
-    },
-    {"version", version, METH_NOARGS,
-     "version()\n--\n\n"
-     "Return the qhull version string."},
-    {NULL, NULL, 0, NULL}
-};
-
-static struct PyModuleDef qhull_module = {
-    PyModuleDef_HEAD_INIT,
-    "qhull", "Computing Delaunay triangulations.\n", -1, qhull_methods
-};
-
-#pragma GCC visibility push(default)
-
-PyMODINIT_FUNC
-PyInit__qhull(void)
-{
-    import_array();
-    return PyModule_Create(&qhull_module);
+PYBIND11_MODULE(_qhull, m) {
+    m.doc() = "Computing Delaunay triangulations.\n";
+
+    m.def("delaunay", &delaunay,
+        "delaunay(x, y, /)\n"
```

Comment:
You can use https://pybind11.readthedocs.io/en/stable/basics.html instead to specify argument names (possibly together with https://pybind11.readthedocs.io/en/stable/reference.html?highlight=kw_only#_CPPv48pos_only if you want to keep them positional-only).

Reply:
OK, done this. It does improve the docstring.